File: common/src/main/java/scooterdelta/challenge/bot/common/domain/Cell.java
package scooterdelta.challenge.bot.common.domain;

import java.util.Objects;

public class Cell extends AbstractCell {

    private boolean occupied;
    private boolean hit;

    public boolean isOccupied() {
        return occupied;
    }

    public void setOccupied(final boolean occupied) {
        this.occupied = occupied;
    }

    public boolean isHit() {
        return hit;
    }

    public void setHit(final boolean hit) {
        this.hit = hit;
    }

    @Override
    public String toString() {
        return "Cell{" +
                "occupied=" + occupied +
                ", hit=" + hit +
                '}';
    }

    @Override
    public boolean equals(final Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        if (!super.equals(o)) return false;
        final Cell cell = (Cell) o;
        return occupied == cell.occupied &&
                hit == cell.hit;
    }

    @Override
    public int hashCode() {
        return Objects.hash(super.hashCode(), occupied, hit);
    }
}

File: common/src/main/kotlin/scooterdelta/challenge/bot/common/state/remote/domain/Cell.kt
package scooterdelta.challenge.bot.common.state.remote.domain

import com.fasterxml.jackson.annotation.JsonProperty

class Cell(

        @JsonProperty("X")
        x: Int,

        @JsonProperty("Y")
        y: Int,

        @JsonProperty("Occupied")
        private val occupied: Boolean,

        @JsonProperty("Hit")
        private val hit: Boolean,

        @JsonProperty("Shielded")
        private val shielded: Boolean,

        @JsonProperty("ShieldHit")
        private val shieldHit: Boolean

) : BaseCell(x, y) {
    override fun printContent(): String {
        return when {
            occupied -> {
                "+"
            }
            hit -> {
                "X"
            }
            shieldHit -> {
                "@"
            }
            shielded -> {
                "%"
            }
            else -> {
                "0"
            }
        }
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other?.javaClass != javaClass) return false
        if (!super.equals(other)) return false

        other as Cell

        if (occupied != other.occupied) return false
        if (hit != other.hit) return false

        return true
    }

    override fun hashCode(): Int {
        var result = super.hashCode()
        result = 31 * result + occupied.hashCode()
        result = 31 * result + hit.hashCode()
        return result
    }

    override fun toString(): String {
        return "Cell(occupied=$occupied, hit=$hit) ${super.toString()}"
    }

}
